# 1.Complete the following table for Water:
| T, degC     | P, kPa      |u, kJ/kg  | Phase description|
| ----------- | ----------- |----------|------------------|
|             | 560         | 1470     |                  |
| 240   |         |          |       Saturated vapor           |
| 150   |      2600   |          |                   |
|    |   4200      |     3100     |                  |



Lookup tables from the e-textbook: https://pressbooks.bccampus.ca/thermo1/chapter/thermodynamic-tables/

In [1]:
import CoolProp.CoolProp as CP

In [2]:
###========== (a)================###
P = 0.56e6 # in Pa
u = 1470 # in kJ/kg
fluid = "water"
uf = CP.PropsSI("U", "Q", 0, "P", P,fluid)/1e3
ug = CP.PropsSI("U", "Q", 1, "P", P, fluid)/1e3
print("Uf at given pressure: {} kJ/kg".format(round(uf,2)))
print("Ug at given pressure: {} kJ/kg".format(round(ug,2)))
## we are in saturated mixture region; since u_given > uf and u_given < ug!
T = CP.PropsSI("T","P",P,"Q",0,fluid)
print("Phase description: Saturated mixture")
print("Temperature: {}".format(round(T - 273.15,2)),"deg C")

Uf at given pressure: 658.15 kJ/kg
Ug at given pressure: 2564.51 kJ/kg
Phase description: Saturated mixture
Temperature: 156.15 deg C


In [3]:
###========== (b)================###
fluid = "water"
T = 240 + 273.15 # Kelvin
uf = CP.PropsSI("U", "Q", 0, "T", T,fluid)/1e3
ug = CP.PropsSI("U", "Q", 1, "T", T, fluid)/1e3
print("U at given temperature : {} kJ/kg".format(round(ug,1)))
pressure = CP.PropsSI("P", "T", T, "Q", 1, fluid)
print("Pressure: {} kPa".format(round(pressure /1e3,1)))

U at given temperature : 2603.1 kJ/kg
Pressure: 3346.9 kPa


In [4]:
###========== (c) ================###
fluid = "water"
T = 150 + 273.15 # Kelvin
P = 2600e3 # Pa
Psat = CP.PropsSI("P", "T", T, "Q",0,fluid)/1e3
## P > Psat; hence compressed fluid
print("Phase description: Compressed liquid")
u_state = CP.PropsSI("U", "T", T, "P", P, fluid)
uf = CP.PropsSI("U", "T", T, "Q", 0, fluid)
print("u = {} kJ/kg".format(round(u_state/1e3,2)))
print("uf  = {} kJ/kg".format(round(uf/1e3,2)))

Phase description: Compressed liquid
u = 630.66 kJ/kg
uf  = 631.66 kJ/kg


In [5]:
###========== (d)================###
fluid = "water"
pressure = 4200e3 # Pa
u_given_state = 3100e3 # in J/kg
# calculating u at q = 0 (sat. liquid)
## add comments to help the audience to help them to understand the arguments that you supply
uf = CP.PropsSI("U", "Q", 0, "P", pressure,fluid)/1e3
ug = CP.PropsSI("U", "Q", 1, "P", pressure, fluid)/1e3
print("Uf at given pressure: {} kJ/kg".format(round(uf,1)))
print("Ug at given pressure: {} kJ/kg".format(round(ug,1)))
## we are in ssuperheated vapor region; u > ug!
T = CP.PropsSI("T", "P", pressure, "U", u_given_state,fluid)
print("Temperature at the final state in deg C: {}".format(round(T-273.15,2)))

Uf at given pressure: 1096.4 kJ/kg
Ug at given pressure: 2601.0 kJ/kg
Temperature at the final state in deg C: 500.8


## Completed table:
| T, degC     | P, kPa      |u, kJ/kg  | Phase description|
| ----------- | ----------- |----------|------------------|
|      **156.15**       | 560         | 1470     |      **Saturated liq-vapour mixture**            |
| 240   |   **3346.92**      |     **2603.12**     |       Saturated vapor           |
| 150   |      2600   | **630.66**         |  **Compressed Liquid**                 |
| **500.8**   |   4200      |     3100     |         **Superheated vapor**         |

